content/blog/lift-and-transform-apps-with-hpe-csi-driver-for-kubernetes.md

```diff
@@ -0,0 +1,283 @@
+
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: hpe-transform
+provisioner: csi.hpe.com
+parameters:
+  csi.storage.k8s.io/fstype: xfs
+  csi.storage.k8s.io/controller-expand-secret-name: hpe-backend
+  csi.storage.k8s.io/controller-expand-secret-namespace: hpe-storage
+  csi.storage.k8s.io/controller-publish-secret-name: hpe-backend
+  csi.storage.k8s.io/controller-publish-secret-namespace: hpe-storage
+  csi.storage.k8s.io/node-publish-secret-name: hpe-backend
+  csi.storage.k8s.io/node-publish-secret-namespace: hpe-storage
+  csi.storage.k8s.io/node-stage-secret-name: hpe-backend
+  csi.storage.k8s.io/node-stage-secret-namespace: hpe-storage
+  csi.storage.k8s.io/provisioner-secret-name: hpe-backend
+  csi.storage.k8s.io/provisioner-secret-namespace: hpe-storage
+  description: Volume transformed from vVol by HPE CSI Driver
+  allowOverrides: importVolAsClone,importVolumeName,forceImport
+reclaimPolicy: Delete
+allowVolumeExpansion: true
+```
+
+The important details in the `StorageClass` are the `allowOverrides`. Here we put the parameters we want the application admin to override once testing has started (`importVolAsClone`) and once the database is ready for the final import (`importVolumeName` and `forceImport`).
+
+For the final production instance, we’re going to use the [Bitnami MariaDB Helm chart](https://artifacthub.io/packages/helm/bitnami/mariadb). For this chart to install cleanly and be able to bring up the database, the following considerations needs to be taken into account:
+
+* There needs to be a known `root@localhost` account with a known password in the source database.
+* The Bitnami chart does not support altering the `mountPath`. The source filesystem needs to have a symlink to `/var/lib/mysql` from `/var/lib/data`.
+
+For the initial cloning of the source database, the following `values.yaml` file is being used.
+
+```yaml
+image:
+  tag: 10.3
+auth:
+  rootPassword: my-password
+global:
+  imagePullSecrets: # Only needed if you're rate limited to Docker Hub
+  - regcred
+primary:
+  persistence:
+    enabled: true
+    storageClass: hpe-transform
+    annotations:
+      csi.hpe.com/importVolAsClone: virt-mm-db-1.vmdk
+    size: 64Gi
+    subPath: lib
+```
+
+The above parameters are quite self-explanatory, the `subPath` directive instructs the kubelet to mount the `lib` directory in the root of the volume. As a result, `/lib` in the volume will be mounted at `/bitnami/mariadb` and the `data` subdirectory which hold the MariaDB databases will be seen properly by the container binaries.
+
+Deploying the chart is straight forward.
+
+```markdown
+$ helm repo add bitnami https://charts.bitnami.com/bitnami
+$ helm install my-test-clone bitnami/mariadb -f values.yaml
+```
+
+Once the chart has been deployed and the database has come up, the “Hello World” test would of course be verifying the database.
```

Comment:
Make this sentence active as opposed to passive:
"Once the chart is deployed and the database comes up, the "Hello World" test will, of course, verify the database."